### ISSUE TITLE
VPN documentation

### ISSUE THREAD

**Issue by Forage** [Catch-up-TV-and-More/plugin.video.catchuptvandmore#288]
Hi,

I wanted to give the VPN feature a try today, but it's not clear to me how to use it, nor is it documented somewhere.

I imported a config file, but the only thing I get is an error message when hitting "connect" saying "an error has occurred while trying to connect OpenVPN".
Which doesn't surprise me that much since at no point I've asked for user name and password.

How am I supposed to use this feature? 
Where can I provide user name and password for a connection?
What can be supplied how on the additional arguments?

P.s. the dialog popping up after supplying a config file or when supplying arguments could use a label. Especially the one after the config file is confusing without it as you don't know what's being asked for.

**Comment by SylvainCecchetto** [comment]
Hi Forage,

You are right concerning the usage: You need to import a config file. After that you only need to click on Connect from the context menu.
Kodi will only call then `openvpn` binary with your config file as argument.
Currently you can not enter a username or a password. Maybe can you fill them directly in the config file? Or you can add any argument from the OpenVPN tab in the settings.

If you want to debug this part, you can try to connect on your VPN from SSH with this command `openvpn my_config.ovpn`, if it's work, you can use this config file with Kodi.

I never use this feature, but I will try to dust off some confusing labels ;-)
Also, an "How-to" on the Wiki or the website would be good :-P

Let us know if we can help you!

**Comment by Forage** [comment]
Thanks for the info.
I managed to get it up and running.

If a VPN connection requires a username and password one can:

1. Provide an argument `--auth-user-pass /storage/.kodi/userdata/vpn.pass`
2. Or adapt the VPN config file to include the setting `auth-user-pass /storage/.kodi/userdata/vpn.pass`

In both cases pointing to a text file containing the credentials on two lines
```
[username]
[password]
```
All but ideal to store that stuff in a file, but at least it's working.

The testing by SSH helped to track down an issue in my config file where the `up` and `down` settings were pointing to an OpenVPN file that does not exist on the LibreElec platform. Commenting out the two lines did the trick.

Also thanks for mentioning the context menu. I wouldn't have found that way of connecting since it depends on the area where you are for it to show up, which is not the case when you work from favourites links. This makes it a bit cumbersome to connect sometimes. Would it be an idea and possible to have to option presented to connect when you run into a Geo-blocking error?

**Comment by romdudu60** [comment]
Hi,
I am confused about VPN, I purchased IPVanish yesterday and installed today inside LibreELEC with following instructions.
https://support.ipvanish.com/hc/en-us/articles/115002080053-Setting-Up-OpenVPN-on-LibreELEC-Kodi
From this point, should Catch-Up TV and More be ready to watch Japanese Replay? (my main goal using VPN). Iget an error 403 on TV Asahi for exemple.
Should I use OpenVPN parameters provided by CatchUp or Zomboided add-on is enough?
I am very new in LibreELEC, Kodi,... Discussions above is right now difficult to understand for me as I don't know about SSH. 
Thanks in advance,

**Comment by Forage** [comment]
Catch-up-TV does not use the "VPN Manager" add-on, it has it's own config for that.

We can't provide support for that add-on. I can't even tell you how it's supposed to work and if it will provide a VPN connection for you whole system. If this should be the case you'd need to find support elsewhere.

The alternative is to use the VPN config options in the Catch-up-TV  add-on as explained above.

**Comment by romdudu60** [comment]
Thank you very much for your reply. I will now try to configure it.

**Comment by romdudu60** [comment]
I tried VPN setting inside Catch-Up TV thanks to @Forage post (--auth-user-pass argument) and managed to connect to Tokyo's server. But I get the same 403 error when trying to watch TV Asahi replay. 
I also tried on Raspbian, i was able to check my IP with internet (looks good). I've tried specifing IPVanish recommanded DNS. I tried changing date localization. I assigned static IP to Raspberry Pi inside DHCP parameters. >Always the same error.
If I can make some diagnostic or anything to understand, please tell me. 

Except this VPN issue, I have to say that this add-on is very good to watch French TV or replay (no VPN). Thank you all.

**Comment by romdudu60** [comment]
Hi,
I don't have anymore this error 403. Looks that probleme was the VPN (or my bad usage). I am now using a VPN made in Japan and it display the name of the file on the top right instead of error 403. 
But stream is not displaying, log indicates OpenDemusStream - Error creating demuxer.
Of course it's not the good place to discuss about this. Any help would be welcome.

**Comment by wwark** [comment]
Hi,
They changed something on the m3u8 format and kodi couldn't play the format ...
I switched to another format, the stream play now but we lost the download mode.
The change is available on 0.2.22~beta02.

**Comment by romdudu60** [comment]
Hi, my issue is SOLVED!
It's working now. @SylvainCecchetto , @Forage , @wwark , my wife and my sons can now watch Japanese replays (NHK, ...) on LibreELEC (Rpi4). You can understand how great it is for our French/ Japanese family. Merci beaucoup à vous et à tous ceux qui y contribuent.
After installing 0.2.22~beta02, Error 403 came back, I re-installed Zomboided VPN Manager Addon and got it working. From my little experience, when VPN errors comes up, it's easier to fix with Zomboided VPN Manager that gives more information about what's wrong. 
I you want me to make some tests to understand what happened. Let me know.

**Comment by wwark** [comment]
Yes for me it helps for my skill of Japanese language and I worked few months in Tokyo.
For HTTP 403 we can't do anything it is a VPN issue.
The VPN can be detected by the channel and blocked your access.
Regards,

**Comment by joaopa00** [comment]
Is this problem still relevant?
